frontend/packages/console-app/src/components/cloud-shell/CloudShellMastheadButton.tsx

```diff
@@ -21,17 +22,41 @@ type StateProps = {
 type Props = WithFlagsProps & StateProps & DispatchProps;
 
 const ClouldShellMastheadButton: React.FC<Props> = ({ flags, onClick, open }) => {
+  const [terminalAvailable, setTerminalAvailable] = React.useState(false);
+  const flagEnabled = flags[FLAG_DEVWORKSPACE];
+  React.useEffect(() => {
+    let mounted = true;
+    if (flagEnabled) {
+      checkTerminalAvailable()
+        .then(() => {
+          if (mounted) {
+            setTerminalAvailable(true);
+          }
+        })
+        .catch(() => {
+          if (mounted) {
+            setTerminalAvailable(false);
+          }
+        });
+    } else {
+      setTerminalAvailable(false);
```

Comment:
Need a check for mounted here as well.

Reply:
@rohitkrai03 this doesn't require a check because it's not executed asynchronously.

Reply:
Ah, right.